src/i_sndfile.c

```diff
@@ -521,35 +521,38 @@ static boolean OpenFile(sndfile_t *file, void *data, sf_count_t size)
     return true;
 }
 
-static void FadeInMono16(short *data, ALsizei size, ALsizei freq)
+static void FadeInOutMono16(short *data, ALsizei size, ALsizei freq)
 {
     const int fadelen = freq * FADETIME / 1000000;
     int i;
 
-    if (!data[0] || size / sizeof(short) < fadelen)
+    if ((!data[0] && !data[size - 1]) || size / sizeof(short) < fadelen)
```

Comment:
I get a "wild pointer" error here and below. Need to set:
```c
const int len = size / sizeof(short);
````
and use
```c
if ((!data[0] && !data[len - 1]) || len < fadelen)
```
and so on.

Reply:
Maybe this should be passed as an argument for size? I haven't worked it out:

https://github.com/fabiangreffrath/woof/blob/f7b5cd317cf296d600b71acbaa6d134f0aa6fe6a/src/i_sndfile.c#L570